ravif/src/av1encoder.rs

```diff
@@ -199,13 +211,11 @@ impl Encoder {
     ///
     /// If all pixels are opaque, the alpha channel will be left out automatically.
     ///
-    /// This function takes 8-bit inputs, but will generate an AVIF file using 10-bit depth.
-    ///
     /// returns AVIF file with info about sizes about AV1 payload.
-    pub fn encode_rgba(&self, in_buffer: Img<&[rgb::RGBA<u8>]>) -> Result<EncodedImage, Error> {
+    pub fn encode_rgba<P: Pixel + Default>(&self, in_buffer: Img<&[Rgba<P>]>) -> Result<EncodedImage, Error> {
         let new_alpha = self.convert_alpha(in_buffer);
-        let buffer = new_alpha.as_ref().map(|b| b.as_ref()).unwrap_or(in_buffer);
-        let use_alpha = buffer.pixels().any(|px| px.a != 255);
+        let buffer = new_alpha.as_ref().map(|b: &Img<Vec<Rgba<P>>>| b.as_ref()).unwrap_or(in_buffer);
+        let use_alpha = buffer.pixels().any(|px| px.a != P::cast_from(255));
```

Comment:
`cast_from` doesn't seem to be applying any scaling. For 10-bit depth, the opaque alpha is 1023, so I don't see how this would work.

Reply:
This change only changes the type to `rav1e::Pixel` since that is what is eventually expected anyways. I wanted to keep the alpha changes for a separate PR to keep this reviewable, this doesn't change the functionality. Hardcoding this based on the format didn't seem like a great solution to me since 12 bit values are also valid.